docs/development/extensions-contrib/spectator-histogram.md

```diff
@@ -0,0 +1,453 @@
+| type      | Must be "longSum", "doubleSum", or "floatSum".                                 | yes       |
+| name      | A String for the output (result) name of the aggregation.                      | yes       |
+| fieldName | A String for the name of the input field containing pre-aggregated histograms. | yes       |
+
+## Post Aggregators
+
+### Percentile (singular)
+This returns a single percentile calculation based on the distribution of the values in the aggregated histogram.
+
+```
+{
+  "type": "percentileSpectatorHistogram",
+  "name": <output name>,
+  "field": {
+    "type": "fieldAccess",
+    "fieldName": <name of aggregated SpectatorHistogram>
+  },
+  "percentile": <decimal percentile, e.g. 50.0 for median>
+}
+```
+
+| Property   | Description                                                 | Required? |
+|------------|-------------------------------------------------------------|-----------|
+| type       | This String should always be "percentileSpectatorHistogram" | yes       |
+| name       | A String for the output (result) name of the calculation.   | yes       |
+| field      | A field reference pointing to the aggregated histogram.     | yes       |
+| percentile | A single decimal percentile between 0.0 and 100.0           | yes       |
+
+### Percentiles (multiple)
+This returns an array of percentiles corresponding to those requested.
+
+```
+{
+  "type": "percentilesSpectatorHistogram",
+  "name": <output name>,
+  "field": {
+    "type": "fieldAccess",
+    "fieldName": <name of aggregated SpectatorHistogram>
+  },
+  "percentiles": [25, 50, 75, 99.5]
+}
+```
+
+> Note: It's more efficient to request multiple percentiles in a single query
+than to request individual percentiles in separate queries. This array-based
+helper is provided for convenience and has a marginal performance benefit over
+using the singular percentile post-aggregator multiple times within a query.
+The more expensive part of the query is the aggregation of the histogram.
+The post-aggregation calculations all happen on the same aggregated histogram.
+
+Results will contain arrays matching the length and order of the requested
+array of percentiles.
+
+```
+"percentilesAdded": [
+    0.5504911679884643, // 25th percentile
+    4.013975155279504,  // 50th percentile 
+    78.89518317503394,  // 75th percentile
+    8580.024999999994   // 99.5th percentile
+]
+```
+
+| Property    | Description                                                  | Required? |
+|-------------|--------------------------------------------------------------|-----------|
+| type        | This String should always be "percentilesSpectatorHistogram" | yes       |
+| name        | A String for the output (result) name of the calculation.    | yes       |
+| field       | A field reference pointing to the aggregated histogram.      | yes       |
+| percentiles | Non-empty array of decimal percentiles between 0.0 and 100.0 | yes       |
+
+## Appendix
```

Comment:
We don't typically include an appendix in docs. Consider making this "Examples" and promoting the heading for bucket boundaries.
```suggestion
## Examples
```